scripts/app.js

```diff
@@ -18,56 +18,61 @@ Project.prototype.toHtml = function(){
     $('<a>').attr('id', 'table-of-contents-link').text(this.title)));
   return templateRender(this);
 };
+// AJAX/JSON UGHHHH
+// ==============================================
+Project.fetchAll = function(){
+  console.log('calling');
+  if(localStorage.rawData){
+    let rawData = JSON.parse(localStorage.rawData);
+    rawData.forEach(function(projectObject){
+      projects.push(new Project(projectObject));
+    });
+    projects.forEach(function(addedProject){
+      $('#projects-here').append(addedProject.toHtml());
+    });
+    Project.addAllHandlers();
+  }else{
+    $.getJSON('/data/rawData.json')
+    //if success:
+    .then(function(data){
+      console.log(data);
+      data.forEach(function(projectObject){
+        projects.push(new Project(projectObject));
+      });
```

Comment:
Your .done is really great! It is great you are exploring the code and chaining!